### PR TITLE
Added additional crypto

### PR DESCRIPTION
**Pull request by Mikerah**
I added the necessary crypto needed to make this work with discv5.
Here's the motivation for making this change: https://github.com/ChainSafe/libp2p-discv5/issues/21

### CODE REVIEW

**Comment by GregTheGreek** [comment]
@Mikerah Your build is failing on CI

**Comment by GregTheGreek** [comment]
Still failing

**Comment by Mikerah** [comment]
@GregTheGreek I know. I tried changing the commit messages as you can tell but I ended up somehow adding to it and now this PR has way to many commits. I think I might have to manually change the remaining commit messages by hand.

**Comment by GregTheGreek** [comment]
@Mikerah you don't need to change commit messages, its just tests failing..

**Comment by Mikerah** [comment]
Both tests and the commitlint are failing from what I can tell. Let me try again.

**Comment by GregTheGreek** [comment]
@Mikerah negative, please reference the `branch` CI

**Comment by Mikerah** [comment]
@GregTheGreek the `branch` CI passes but the `Pull Request` CI still fails.

**Comment by mpetrunic** [comment]
> @GregTheGreek the `branch` CI passes but the `Pull Request` CI still fails.

Pull Request is more important CI. Merge/rebase master here and check why its failing

**Comment by Mikerah** [comment]
@mpetrunic It's failing because of the commitlint messages. If you look at the commit history, you'll noticed I made changes to the the commit messages. However, it doesn't seem to have affected the pass commit messages as I would expect.

**Comment by mpetrunic** [comment]
> @mpetrunic It's failing because of the commitlint messages. If you look at the commit history, you'll noticed I made changes to the the commit messages. However, it doesn't seem to have affected the pass commit messages as I would expect.

This will fail if we merge to master so this need to be fixed.

Either look into history rewriting and commit everything again properly or cherry pick with proper commit messages to another branch. If there isn't much changes you could manually extract them and commit again in fresh branch

**Comment by GregTheGreek** [comment]
@Mikerah there are still merge conflicts

**Comment by Mikerah** [comment]
@GregTheGreek all checks pass

**Review by wemeetagain — APPROVED** [review]
I think upstreaming this work would be preferable to us maintaining our own fork. The js-libp2p ecosystem could benefit from additional crypto as well as us.

**Comment by Mikerah** [comment]
@GregTheGreek or @mpetrunic I need another review on this PR.

**Review by mpetrunic — APPROVED** [review]
